app.go

```diff
@@ -135,6 +135,13 @@ func (s *App) Stop(ctx context.Context) error {
 	return withTimeout(ctx, s.lifecycle.stop)
 }
 
+// Done allows blocking on SIGINT or SIGTERM
+func (App) Done() <-chan os.Signal {
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, syscall.SIGINT, syscall.SIGTERM)
+	return c
+}
```

Comment:
Train is shoveling coal right now, ready to depart. Haven't left yet though!

Could you elaborate? We'd still listen on signals notify via empty struct?
How does that allow us to future-proof?